FrontPage/ServiceProvider.swift

```diff
@@ -0,0 +1,126 @@
+//
+//   - Hardcoding the link to the featured shows rather than taking it from emergence
+//   - Linking to Pod's framework for Keys/Auth directly, rather than letting CP do it properly
+//     framework paths / CP are using relative paths, that should be fine.
+//
+// Other than that, not too bad. ./
+
+// So this is an OS extension ran by the OS when we send a notification with the string
+// TVTopShelfItemsDidChangeNotification - which we do on launch in the app
+
+// For debugging: you run the extension then launch emergence, it won't load properly
+//                so you then switch targets to Emergence and your app will run in the
+//                background. Breakpoints/Exceptions will work then :+1:.
+
+class ServiceProvider: NSObject, TVTopShelfProvider {
+
+    var topShelfStyle: TVTopShelfContentStyle {
+        return .Sectioned
+    }
+
+    lazy var featuredShowsWrapper: TVContentItem = {
+        let wrapperID = TVContentIdentifier(identifier: "featured-shows", container: nil)!
+        let wrapperItem = TVContentItem(contentIdentifier: wrapperID)!
+        wrapperItem.title = "Featured Shows"
+        return wrapperItem
+    }()
+
+    func auth(completion: ((ArtsyToken!, NSError!) -> Void)) -> (){
+        let keys = EmergenceKeys()
+        let auth = ArtsyAuthentication(clientID: keys.artsyAPIClientKey(), clientSecret: keys.artsyAPIClientSecret())
+        auth.getWeekLongXAppTrialToken(completion)
+    }
+
+    var topShelfItems: [TVContentItem] {
+        var items:[TVContentItem] = []
+        let semaphore = dispatch_semaphore_create(0)
+
+        auth { token, error in
+            // No auth, no go
+            if error != nil {
+                dispatch_semaphore_signal(semaphore)
+                return print("Error: \(error)")
+            }
+
+            // Setup the networking stack
+            let networkOp = ArtsyNetworkOperator()
+            let url = NSURL(string: "https://api.artsy.net/api/v1/set/530ebe92139b21efd6000071/items")!
+
+            // Make sure we apply our new xapp token
+            let request = NSMutableURLRequest(URL: url)
+            request.setValue(token.token, forHTTPHeaderField: "X-Xapp-Token")
+
+            let _ = networkOp.JSONTaskWithRequest(request, success: { request, response, json in
+
+                // When this block wraps up we can say we're done
+                defer { dispatch_semaphore_signal(semaphore) }
+
+                // This is a collection of shows raw JSON
+                guard let showDicts = json as? [[String:AnyObject]] else {
+                    return print("Could not convert the JSON to an array")
+                }
+
+                self.featuredShowsWrapper.topShelfItems = showDicts.map(self.showDictToWrapper)
+                items = [self.featuredShowsWrapper]
+
+            }, failure: { (request, response, error, json) in
+
+                // Less OK, but can't do much atm
+                dispatch_semaphore_signal(semaphore)
```

Comment:
well, less OK means that the itemset has probably been deleted, a bit out of our control really